pkg/providers/aws/provider_redis.go

```diff
@@ -587,3 +616,56 @@ func (p *AWSRedisProvider) setRedisServiceMaintenanceMetric(ctx context.Context,
 	}
 	return nil
 }
+
+// CreateElastiCacheAvailabilityAlert Call this when we create the ElastiCache instance to create an
+// alert to watch for the availability of the instance
+func (p *AWSRedisProvider) CreateElastiCacheAvailabilityAlert(ctx context.Context, resourceID string, namespace string, instanceID string, clusterID string) error {
+	alertRuleName := fmt.Sprintf("cro-aws-elasticache-%s", instanceID)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("absent(cro_aws_elasticache_available{namespace='%s',instanceID='%s',clusterID='%s',resourceID='%s'} == 1)",
+			namespace, instanceID, clusterID, resourceID),
+	)
+
+	pr, err := croResources.CreatePrometheusRule(namespace, alertRuleName, alertExp)
+	if err != nil {
+		return err
+	}
+
+	// Unless it already exists, call the kubernetes api and create this PrometheusRule
+	// Replace this with CreateOrUpdate if we can figure it out
+	err = p.Client.Create(ctx, pr)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			msg := fmt.Sprintf("exception calling Create metricName: %s", alertRuleName)
```

Comment:
as we are only using the string once, I think we can return the `Sprintf` directly in the error